posts/2022-08-04-jakarta-core-profile-22009-beta.adoc

```diff
@@ -105,64 +105,121 @@ You can find out more about this specific update and the wider technology with t
 
-In previous versions of CDI, a non-empty `beans.xml` file that did not include a version attribute defaulted to mean `bean-discovery-mode="all"`. In CDI 4.0, it defaults to `bean-discovery-mode="annotated"`.
+=== Unversioned beans.xml files
 
-Two new Observable Container State Events are added: Startup and Shutdown. Applications can listen for these events to be notified when the CDI container is starting up and being shut down.
+In previous versions of CDI, a non-empty `beans.xml` file that did not include a `version` attribute defaulted to `bean-discovery-mode="all"`. In CDI 4.0, it defaults to `bean-discovery-mode="annotated"`. It is recommended that all `beans.xml` files should be properly versioned.
 
-A new Handle API is added to make programatic lookup of Beans and Instances easier.
+[source, xml]
+----
+<?xml version="1.0" encoding="UTF-8"?>
+<beans xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xmlns="https://jakarta.ee/xml/ns/jakartaee"
+    xsi:schemaLocation="https://jakarta.ee/xml/ns/jakartaee https://jakarta.ee/xml/ns/jakartaee/beans_4_0.xsd"
+    version="4.0"
+    bean-discovery-mode="annotated">
+</beans>
+----
 
-In Jakarta Annotations 2.1, the `@priority` annotation can now be applied to any program element. This annotation is used by CDI 4.0 to allow Stereotypes to include a Priority annotation.
+=== Programatic lookup of Beans and Instances
 
-The following previously deprecated CDI APIs are removed:  
+A new Handle API is added to make programatic inspection of Bean metadata easier, avoiding the need to create instances before they are actually required. `Handles` may be obtained via the `Instance` API.
 
-* `@New qualifier`, which is replaced by `@dependent` beans in CDI 1.1 and later.
+[source, java]
+----
+public interface Handle<T> extends AutoCloseable {
+        T get();
+        Bean<T> getBean();
+        void destroy();
+        void close();
+}
+----
 
-* `Bean#isNullable()`, which has not been used by the implementation since CDI 1.1.
+=== Removed APIs
 
-* `BeanManager#fireEvent()`. Use `BeanManager.getEvent()` instead in CDI 2.0 and later. 
+The following previously deprecated CDI APIs have been removed:  
 
-* `BeanManager#createInjectionTarget(AnnotatedType)`, which is replaced by `BeanManager#getInjectionTargetFactory(AnnotatedType)` in CDI 1.1 and later.
+* `@New` qualifier
+  - replaced by `@Dependent` beans
+* `Bean#isNullable()`
+  - has not been used by the implementation since CDI 1.1
+* `BeanManager#fireEvent()`
+  - Use `BeanManager.getEvent()` instead
+* `BeanManager#createInjectionTarget(AnnotatedType)`
+  - replaced by `BeanManager#getInjectionTargetFactory(AnnotatedType)`
+* `BeforeBeanDiscovery#addAnnotatedType(AnnotatedType)`
+  - replaced by `BeforeBeanDiscovery#addAnnotatedType(AnnotatedType, String)`
```

Comment:
Is there a reason for saying "Use.... instead" in some cases, but "replace by ...." in others? Or could they all be worded as "Use....instead"?